nestkernel/conn_builder.cpp

```diff
@@ -1101,14 +1101,28 @@ nest::FixedInDegreeBuilder::connect_()
       {
         // check whether the target is on this mpi machine
         if ( not kernel().node_manager.is_local_gid( *tgid ) )
+        {
+          for ( long j = 0; j < indegree_; ++j )
+          {
+            // skip array parameters handled in other virtual processes
+            skip_conn_parameter_( tid );
```

Comment:
Good idea. I've done it. However I also noticed that the same is true for the connection rule all_to_all
in conn_builder.cpp, line 849-853:

          for ( GIDCollection::const_iterator sgid = sources_->begin();
                sgid != sources_->end();
                ++sgid )
            skip_conn_parameter_( tid );

and same thing in other parts of the same file. Since the number of source neurons is fixed in the all_to_all rule, also in this case it should be possible to avoid those loops and to pass the number of source neurons as second argument of the call to skip_conn_parameter